File: app/src/main/java/ru/geekbrains/lessions2345/notepadonfragments/logic/CardSource.java
package ru.geekbrains.lessions2345.notepadonfragments.logic;

public interface CardSource {
    int size();

    CardNote getCardNote(int position);

    ListNotes getListNotes();

    void addCardNote();

    void addCardNote(CardNote cardNote);

    void setCardNote(int position, CardNote cardNote);
    void setCardNote(int position, String name, String description);
    void setCardNote(int position, String text);
    void setCardNote(int position, int year, int month, int day);
    void setCardNote(boolean deleteMode, int activeNoteIndex, boolean createNewNoteMode, int oldActiveNoteIndexBeforeDelete);

    void removeCardNote(int position);
}

File: app/src/main/java/ru/geekbrains/lessions2345/notepadonfragments_2/logic/CardSource.kt
package ru.geekbrains.lessions2345.notepadonfragments_2.logic

interface CardSource {
    fun size(): Int

    fun getCardNote(position: Int): CardNote

    fun getListNotes(): ListNotes

    fun addCardNote()

    fun addCardNote(cardNote: CardNote)

    fun setCardNote(position: Int, cardNote: CardNote)

    fun setCardNote(position: Int, name: String, description: String)

    fun setCardNote(position: Int, text: String)

    fun setCardNote(position: Int, year: Int, month: Int, day: Int)

    fun removeCardNote(position: Int)
}